realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1092,20 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Condition that finds a distinct set of objects of a specific class. As a Realm is unordered,
+     * it is undefined which objects are returned in case of multiple occurrences.
+     *
+     * @param fieldName the field name.
+     * @return the query object.
+     * @throws IllegalArgumentException if a field name does not exist or the field is not indexed.
+     */
+    public RealmQuery<E> distinct(String fieldName) {
+        long columnIndices[] = schema.getColumnIndices(fieldName, RealmFieldType.STRING);
+        this.query.distinct(columnIndices);
+        return this;
```

Comment:
I am not sure this will work. I think of `distinct()` as a selector just like `findAll` and `findFirst` it is not something you can append to.